include/pacemaker.h

```diff
@@ -26,6 +26,35 @@ extern "C" {
 
 #  include <crm/stonith-ng.h>
 
+/*!
+ * \brief Control which portions of cluster status are displayed
+ */
+typedef enum {
+    show_stack         = 1 << 0,
+    show_dc            = 1 << 1,
+    show_times         = 1 << 2,
+    show_counts        = 1 << 3,
+    show_options       = 1 << 4,
+    show_nodes         = 1 << 5,
+    show_resources     = 1 << 6,
+    show_attributes    = 1 << 7,
+    show_failcounts    = 1 << 8,
+    show_operations    = 1 << 9,
+    show_fence_failed  = 1 << 10,
+    show_fence_pending = 1 << 11,
+    show_fence_worked  = 1 << 12,
+    show_tickets       = 1 << 13,
+    show_bans          = 1 << 14,
+    show_failures      = 1 << 15
```

Comment:
I like pcmk_show_e being just a bitfield of sections of output to show.  I'd prefer to keep modifiers of that list like you'd find in pe_print_* defined somewhere else.  In fact, I'd prefer to get rid of pcmk_show_inactive and pcmk_show_rscs_by_node, and maybe even some of the fencing options.  These all seem like just modifying the details of sections of output.

Reply:
Maybe new enums for "sections to display" and "display options"? There are only 6 pe_print_* options that shouldn't be deprecated, and it would be nice if libpacemaker didn't expose anything from libpe_status. I.e. move the usable stuff from pe_print_* to the new display options enum, and use that in the internal output code and new public libpacemaker API. The enums would go in libcrmcommon to be usable everywhere.

Reply:
"Sections to display" and "display options" is basically what we have now, with show_opts and print_opts.  We'd just need to straighten it out a bit and move it out of libpe_status.  Are you thinking that pcmk_show_e would move out of include/pacemaker.h, perhaps into include/crm/common/output.h (which doesn't currently exist)?  Perhaps renaming it to something that indicates it's section-related would be useful.  And then, a new display options enum would be created right beside it?

Reply:
Exactly.

I don't have a preference for one enum or two, either works fine for me.

libcrmcommon is the best place for both libpe_status and libpacemaker to have access to them, because my longer-term goal is to merge libpe_status into libcrmcommon, and it would be better if libpacemaker public APIs never used libpe_status types to begin with. Also, pe_print_options now has more deprecated values than not, so I think starting with a fresh enum and deprecating that one entirely (since it would only be used by the deprecated ->print() method at that point) is a good idea.

Reply:
Looking at this a little more in depth... if I were to be completely consistent, either all the fencing flags end up in the pcmk_show_e replacement, or pcmk_show_{stack,dc,times,counts,options} need to end up in the print_opts replacement.  It's probably easiest to leave everything where it is, however.

Maybe a sensible break down is that anything with its own formatted output message stays in the pcmk_show_e replacement, and everything else ends up in the print_opts replacement.

This being public API, I'd like to get it right.

Reply:
Another possibility would be anything that can go in --include/--exclude is in pcmk_show_e, and anything that has its own option goes in the other one.

Reply:
I think that is essentially the same thing.  --include=summary maps to a block of messages, and --include=fencing is kind of tricky in that it both sets several flags that do have their own messages and also causes something special to be done just for the XML formatter.

Reply:
Given that the lines are blurry, now I'm thinking about one enum again, but maybe name the values in a way that points at the relationships, pcmk_show_SECTION_OPTION?

E.g. pcmk_show_summary_all, pcmk_show_summary_stack, pcmk_show_summary_dc, etc. and pcmk_show_resources, pcmk_show_resources_inactive, pcmk_show_resources_by_node, etc.